packages/react-dom/src/client/ReactDOMHostConfig.js

```diff
@@ -366,9 +366,14 @@ export function appendChildToContainer(
   // through the React tree. However, on Mobile Safari the click would
   // never bubble through the *DOM* tree unless an ancestor with onclick
   // event exists. So we wouldn't see it and dispatch it.
-  // This is why we ensure that containers have inline onclick defined.
+  // This is why we ensure that non React root containers have inline onclick
+  // defined.
   // https://github.com/facebook/react/issues/11918
-  if (!container._reactRootContainer && parentNode.onclick === null) {
+  const {_reactRootContainer} = container;
```

Comment:
Another style nit: we don't use destructuring, with some exceptions like module-style objects or function args (component props).